cpp/arcticdb/storage/storage.hpp

```diff
@@ -61,10 +61,12 @@ class NoDataFoundException : public ArcticCategorizedException<ErrorCategory::MI
     std::optional<VariantId> key_;
 };
 
-class KeyNotFoundException : public ArcticSpecificException<ErrorCode::E_DUPLICATE_KEY> {
+class KeyNotFoundException : public ArcticSpecificException<ErrorCode::E_KEY_NOT_FOUND> {
+    using format_string_t = fmt::format_string<Composite<VariantKey>>;
 public:
-    explicit KeyNotFoundException(Composite<VariantKey>&& keys) :
-        ArcticSpecificException<ErrorCode::E_DUPLICATE_KEY>(fmt::format("{}", keys)),
+    explicit KeyNotFoundException(Composite<VariantKey>&& keys,
+                                  format_string_t format = "Not found: {}") :
```

Comment:
I think the API is easier to follow if we just pass the formatted string in here, rather than the `format_string` template

Reply:
When a `KeyNotFoundException` occurs, shouldn't we always ensure that the key is printed too? If you just dump `std::string` given by the user you won't print the key. Or we could modify it as `std::string message = "Not found"`, then format it as `fmt::format(message + ": {}", keys)`. I am asking this because by default, exception output should contain the key to make sense of the exception. So it might be cleaner to use `string_format`.